dali/python/nvidia/dali/pipeline.py

```diff
@@ -453,7 +485,28 @@ def _setup_input_callbacks(self):
             if _is_external_source_with_callback(op):
                 group = op._group
                 groups.add(group)
-        self._input_callbacks = list(groups)
+        groups = list(groups)
+        self._input_callbacks = groups
+        self._parallel_input_callbacks = [group for group in groups if group.parallel]
+        self._seq_input_callbacks = [group for group in groups if not group.parallel]
+
+    def start_py_workers(self):
+        """
+        Start Python workers (that will run ExternalSource callbacks).
+        It's called automatically by Pipeline's build method.
+
+        If you're going to build more than one pipeline that starts Python workers by forking the process
+        (``py_workers_init=fork``) then you need to call ``start_py_workers`` method on all those pipelines before
+        calling build method of any pipeline. This is due to the fact that building the pipeline makes the
+        process acquire CUDA context and forking with CUDA context is unsupported and may lead to unexpected errors.
+        The same applies for using any other functionality that would create CUDA context - you need to call
+        ``start_py_workers`` before you call such functionality when using ``py_workers_init=fork``.
+
+        If you use the method you cannot specify ``define_graph`` when calling ``build``.
```

Comment:
So the classical API with `define_graph` won't work, right?

Reply:
If you refer to overriding pipeline method define_graph, then it should be fine. The only issue is that you can pass define_graph callback to the build method instead. And if you call start_py_workers this part of build method needs to be run before starting workers. So we can either say it is not supported, or allow user to pass such a callback to start_py_workers, which is sort of weird dependency. I talked to Krzysztof and he suggested that it is rare case and we can forbid that. @klecki

Reply:
Sounds legit. I would update the build() method docs as well in such case.

Reply:
done